### DISCUSSION TITLE
Disable running `go mod vendor` entirely

### DISCUSSION THREAD

**Discussion by chancez** [renovatebot/renovate#29107] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I'm using self-hosted renovate, and we have some postUpgradeCommands to manage Go modules/vendor, instead of using the postUpdate options. However, it seems that Renovate will unconditionally try to run `go mod vendor` if it detects a `vendor/` directory. This is causing trouble for us because we've recently adopted using [go workspaces with vendoring](https://github.com/golang/go/issues/60056) which requires using `go work vendor` and fails on `go mod vendor`.

I can't see any way to disable this behavior unfortunately though: https://docs.renovatebot.com/golang/#module-vendoring

### Logs (if relevant)

<details><summary>Logs</summary>

```

         "stack": "ExecError: Command failed: go mod vendor\ngo: 'go mod vendor' cannot be run in workspace mode. Run 'go work vendor' to vendor the workspace or set 'GOWORK=off' to exit workspace mode.\n\n    at ChildProcess.<anonymous> (/usr/local/renovate/lib/util/exec/common.ts:99:11)\n    at ChildProcess.emit (node:events:529:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:292:12)"


```

</details>

**Comment by rarkins** [comment]
Would it be better if renovate ran the go work vendor command instead?

**Reply by chancez** [reply]
@rarkins if a `go.work` is found and a `vendor/` then yeah I think that would be ideal. However, I think it would make sense to have an option to opt out of having renovate run do vendor in case the user has a more complicated requirement and wants to handle it themselves (via postUpgradeCommand)

**Reply by rarkins** [reply]
If you prefer, you can submit a PR adding a "goModVendorSkip" option

**Reply by chancez** [reply]
I opted to do the solution that would help the most people which is to run `go work vendor` as you suggested :) https://github.com/renovatebot/renovate/pull/29216
